utils_cv/detection/model.py

```diff
@@ -373,30 +481,25 @@ def predict_batch(
         image that is scored.
         """
 
-        labels = self.dataset.labels
         model = self.model.eval()
 
         for i, batch in enumerate(dl):
             ims, infos = batch
-            ims = [im.cuda() for im in ims]
+            ims = [im.to(self.device) for im in ims]
             with torch.no_grad():
-                raw_dets = model(list(ims))
-
-            det_bbox_batch = []
-            for raw_det, info in zip(raw_dets, infos):
-
-                im_idx = int(info["image_id"].numpy())
-                im_path = dl.dataset.dataset.im_paths[im_idx]
-
-                det_bboxes = _get_det_bboxes(
-                    [raw_det], labels=labels, im_path=im_path
-                )
-
-                det_bboxes = _apply_threshold(det_bboxes, threshold)
-                det_bbox_batch.append(
-                    {"idx": im_idx, "det_bboxes": det_bboxes}
-                )
-            yield det_bbox_batch
+                raw_dets = model(ims)
+
+            raw_dets = [
```

Comment:
I am a bit at a loss what is done here. The original version was much easier to read. Can you please undo and use an explicit for xxx in zip(zzz) loop again.

Reply:
Yeah, a little bit.  Rewritten.